src/driver.cpp

```diff
@@ -1,113 +1,175 @@
+    this->get_parameter("read_rate", hz_read);
+    this->get_parameter("device_id", device_id);
+    // this->get_parameter("frame_id", frame_id);
+    this->get_parameter("topic_name", topic_name);
+    this->get_parameter("name", name);
+    this->get_parameter("video_stream_recovery_time", m_video_stream_recovery_time);
+    this->get_parameter("video_stream_recovery_tries", m_video_stream_recovery_tries);
+
+
+    // Timers
+    m_proceed_tmr =
+        this->create_wall_timer(std::chrono::milliseconds(int(1000.0 / hz_read)), std::bind(&Driver::proceed, this));
+
+    camera_.reset(new Capture(shared_from_this(), topic_name, PUBLISHER_BUFFER_SIZE, frame_id));
+
+    if (this->get_parameter("file", file_path) && file_path != "")
+    {
+        camera_->openFile(file_path);
+    }
+    else if (this->get_parameter("port", port) && port != "")
     {
-      ROS_WARN("fail to set image_width");
+        if (!camera_->open(port))
+        {
+            RCLCPP_WARN(get_logger(), "Couldnt open camera by port");
+            // return false;
+        }
     }
-  }
-  if (private_node_.getParam("image_height", image_height))
-  {
-    if (!camera_->setHeight(image_height))
+    else
     {
-      ROS_WARN("fail to set image_height");
+        if (!camera_->open(device_id))
+        {
+            RCLCPP_WARN(get_logger(), "Couldnt open camera by device id");
+            // return false;
+        }
     }
-  }
-
-  camera_->setPropertyFromParam(CV_CAP_PROP_POS_MSEC, "cv_cap_prop_pos_msec");
-  camera_->setPropertyFromParam(CV_CAP_PROP_POS_AVI_RATIO, "cv_cap_prop_pos_avi_ratio");
-  camera_->setPropertyFromParam(CV_CAP_PROP_FRAME_WIDTH, "cv_cap_prop_frame_width");
-  camera_->setPropertyFromParam(CV_CAP_PROP_FRAME_HEIGHT, "cv_cap_prop_frame_height");
-  camera_->setPropertyFromParam(CV_CAP_PROP_FPS, "cv_cap_prop_fps");
-  camera_->setPropertyFromParam(CV_CAP_PROP_FOURCC, "cv_cap_prop_fourcc");
-  camera_->setPropertyFromParam(CV_CAP_PROP_FRAME_COUNT, "cv_cap_prop_frame_count");
-  camera_->setPropertyFromParam(CV_CAP_PROP_FORMAT, "cv_cap_prop_format");
-  camera_->setPropertyFromParam(CV_CAP_PROP_MODE, "cv_cap_prop_mode");
-  camera_->setPropertyFromParam(CV_CAP_PROP_BRIGHTNESS, "cv_cap_prop_brightness");
-  camera_->setPropertyFromParam(CV_CAP_PROP_CONTRAST, "cv_cap_prop_contrast");
-  camera_->setPropertyFromParam(CV_CAP_PROP_SATURATION, "cv_cap_prop_saturation");
-  camera_->setPropertyFromParam(CV_CAP_PROP_HUE, "cv_cap_prop_hue");
-  camera_->setPropertyFromParam(CV_CAP_PROP_GAIN, "cv_cap_prop_gain");
-  camera_->setPropertyFromParam(CV_CAP_PROP_EXPOSURE, "cv_cap_prop_exposure");
-  camera_->setPropertyFromParam(CV_CAP_PROP_CONVERT_RGB, "cv_cap_prop_convert_rgb");
-
-  camera_->setPropertyFromParam(CV_CAP_PROP_RECTIFICATION, "cv_cap_prop_rectification");
-  camera_->setPropertyFromParam(CV_CAP_PROP_ISO_SPEED, "cv_cap_prop_iso_speed");
+
+    camera_->setPropertyFromParam(cv::CAP_PROP_POS_MSEC, "cv_cap_prop_pos_msec");
```

Comment:
there are several changes that look like they were caused by the autoformatter. Is very hard to review the PR this way beause we don't know what changes affect the implementation and what are only style. We should keep only the changes that concern the implementation

Reply:
Done